test/jdk/java/awt/Dialog/FileDialogIconTest/FileDialogIconTest.java

```diff
@@ -169,90 +182,103 @@ public void actionPerformed(ActionEvent event) {
 
         setImageButton5.addActionListener(new ActionListener() {
             public void actionPerformed(ActionEvent event) {
-                List<BufferedImage> list = new ArrayList<>();
+                List<Image> list = new ArrayList<>();
                 try {
                     Robot robot = new Robot();
                     Rectangle rectangle;
                     for (int i = 1; i <= 4; i++) {
-                        rectangle = new Rectangle(i * 10, i * 10, i * 10 + 40, i * 10 + 40);
-                        java.awt.image.BufferedImage image = robot.createScreenCapture(rectangle);
+                        rectangle = new Rectangle(i * 10, i * 10,
+                                                  i * 10 + 40, i * 10 + 40);
+                        BufferedImage image = robot.createScreenCapture(rectangle);
                         robot.delay(100);
                         list.add(image);
                     }
+                    setImagesToFrame(list);
                 } catch (Throwable t) {
                     t.printStackTrace();
                 }
-                PassFailJFrame.log("Captured images and set to the list for dialog");
+                PassFailJFrame.log("Captured images and set to the list for frame");
             }
         });
 
         setImageButton6.addActionListener(new ActionListener() {
             public void actionPerformed(ActionEvent event) {
-                List<BufferedImage> list = new ArrayList<>();
+                List<Image> list = new ArrayList<>();
                 try {
                     Robot robot = new Robot();
                     Rectangle rectangle;
                     for (int i = 1; i <= 4; i++) {
-                        rectangle = new Rectangle(i * 10, i * 10, i * 10 + 40, i * 10 + 40);
-                        java.awt.image.BufferedImage image = robot.createScreenCapture(rectangle);
+                        rectangle = new Rectangle(i * 10, i * 10,
+                                                  i * 10 + 40, i * 10 + 40);
+                        BufferedImage image = robot.createScreenCapture(rectangle);
                         robot.delay(100);
                         list.add(image);
                     }
+                    setImagesToFD(list);
```

Comment:
Here, again the code is the same as in `setImageButton5.addActionListener` except for the final call `setImagesToFD` vs `setImagesToFrame`. I suggest moving capture code out of either methods and re-use it.